### ISSUE TITLE
Error when coverage threshold is set in version 5.X

### ISSUE THREAD

**Issue by sridharmallela** [monounity/karma-typescript#369]
With the latest 5.x version, I ran into a problem when tests are passing but throws a generic `karma server unhandled exception error`.  I have added a repo to reproduce this issue [karma-typescript-issue](https://github.com/sridharmallela/karma-typescript-issue)

```js
23 03 2020 02:37:19.750:ERROR [karma-server]: UnhandledRejection
TypeError: Cannot read property 'files' of undefined
    at Threshold.toSummaries (/karma-typescript-issue/node_modules/karma-typescript/src/istanbul/threshold.ts:59:21)
    at Threshold.check (/karma-typescript-issue/node_modules/karma-typescript/src/istanbul/threshold.ts:39:38)
    at Reporter.create.<anonymous> (/karma-typescript-issue/node_modules/karma-typescript/src/karma/reporter.ts:59:46)
    at step (/karma-typescript-issue/node_modules/karma-typescript/dist/karma/reporter.js:33:23)
    at Object.next (/karma-typescript-issue/node_modules/karma-typescript/dist/karma/reporter.js:14:53)
    at fulfilled (/karma-typescript-issue/node_modules/karma-typescript/dist/karma/reporter.js:5:58)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)

```

On further debugging into this issue seems to be related to refactor to adopt to latest istanbul libraries. `istanbul-lib-coverage`.  This is what i found

In [karma/reporter.ts](https://github.com/monounity/karma-typescript/blob/master/packages/karma-typescript/src/karma/reporter.ts)

```
  const coverage = that.coverageMap.get(browser);
  const coverageMap = istanbulCoverage.createCoverageMap();
  coverageMap.merge(coverage);

  const sourceMapStore = istanbulSourceMaps.createSourceMapStore();
  const remappedCoverageMap = await sourceMapStore.transformCoverage(coverageMap);

  if (results && config.hasCoverageThreshold && !threshold.check(browser, remappedCoverageMap.map)) {
    results.exitCode = 1;
  }
```

remappedCoverageMap returns an instance of [istanbul-lib-coverage/coverage-map.js](https://github.com/istanbuljs/istanbuljs/blob/master/packages/istanbul-lib-coverage/lib/coverage-map.js), which doesnot have a property `map` associated.

i have modified this line <strike>`if (results && config.hasCoverageThreshold && !threshold.check(browser, remappedCoverageMap.map)) {`</strike> to `if (results && config.hasCoverageThreshold && !threshold.check(browser, remappedCoverageMap)) {` on my local copy and was able to see expected result.

I am willing to submit a quick PR, if allowed. Thanks!!

**Comment by erikbarke** [comment]
Nice detective work! A PR would more than welcome ☺️
[Contributing guidelines](https://github.com/monounity/karma-typescript/wiki/Contributing)

**Comment by sridharmallela** [comment]
@erikbarke Thanks for response! As the types for `istanbul-lib-source-maps` has been out of sync with source code. I went ahead and submitted a PR to update the types (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43339), once approved and merged will submit the PR.

**Comment by sridharmallela** [comment]
@erikbarke Thanks, i noticed the types for `istanbul-lib-source-maps` is out of sync. i have submitted a PR to address that, once it is approved and merged, I will submit a PR to fix the above issue.

**Comment by sridharmallela** [comment]
PR has been submitted, https://github.com/monounity/karma-typescript/pull/372

**Comment by erikbarke** [comment]
Fixed in #372
